dlt/sources/rest_api/config_setup.py

```diff
@@ -577,45 +646,191 @@ def remove_field(response: Response, *args, **kwargs) -> Response:
+        {"id"}
+    """
+    expressions = set()
+
+    def recursive_search(value: Union[str, List[Any], Dict[str, Any]]) -> None:
+        if isinstance(value, dict):
+            for key, val in value.items():
+                recursive_search(key)
+                recursive_search(val)
+        elif isinstance(value, list):
+            for item in value:
+                recursive_search(item)
+        elif isinstance(value, str):
+            e = [
+                field_name
+                for _, field_name, _, _ in string.Formatter().parse(value)
+                if field_name
+                and (prefixes is None or any(field_name.startswith(prefix) for prefix in prefixes))
+            ]
+            expressions.update(e)
+
+    recursive_search(content)
+    return expressions
+
+
+def _expressions_to_resolved_params(expressions: Set[str]) -> List[ResolvedParam]:
+    resolved_params = []
+    # We assume that the expressions are in the format 'resources.<resource>.<field>'
+    # and not more complex expressions
+    for expression in expressions:
+        parts = expression.strip().split(".", maxsplit=2)
+        if len(parts) != 3:
+            raise ValueError(
+                f"Invalid definition of {expression}. Expected format:"
+                " 'resources.<resource>.<field>'"
+            )
+        resolved_params.append(
+            ResolvedParam(
+                expression,
+                {
+                    "type": "resolve",
+                    "resource": parts[1],
+                    "field": parts[2],
+                },
+            )
+        )
+    return resolved_params
+
+
 def process_parent_data_item(
     path: str,
     item: Dict[str, Any],
     resolved_params: List[ResolvedParam],
-    include_from_parent: List[str],
-) -> Tuple[str, Dict[str, Any]]:
+    params: Optional[Dict[str, Any]] = None,
+    request_json: Optional[Dict[str, Any]] = None,
+    include_from_parent: Optional[List[str]] = None,
+    incremental: Optional[Incremental[Any]] = None,
+    incremental_value_convert: Optional[Callable[..., Any]] = None,
+) -> Tuple[str, Dict[str, Any], Dict[str, Any], Dict[str, Any]]:
+    params_values = collect_resolved_values(
+        item, resolved_params, incremental, incremental_value_convert
+    )
+    expanded_path = expand_placeholders(path, params_values)
+    expanded_params = expand_placeholders(params or {}, params_values)
+    expanded_json = expand_placeholders(request_json or {}, params_values)
+
     parent_resource_name = resolved_params[0].resolve_config["resource"]
+    parent_record = build_parent_record(item, parent_resource_name, include_from_parent)
+
+    return expanded_path, expanded_params, expanded_json, parent_record
 
-    param_values = {}
+
+def convert_incremental_values(
+    incremental: Incremental[Any], convert: Callable[..., Any]
+) -> Dict[str, Any]:
+    return {
+        "incremental.start_value": convert(incremental.start_value),
```

Comment:
IMO you should also convert: `last_value` (backward compatibility, I think this is what we documented before) and `initial_value`

also do we specify that `convert` function must be able to deal with `None`?

Reply:
Implemented 2 in 0c0141d6254bea3bfea27c63f78dc2b9fa1ebc1f